pkg/controllermanager/controller/managedseedset/managedseedset_actuator.go

```diff
@@ -0,0 +1,345 @@
+		a.infoEventf(set, "Creating shoot %s", getFullName(set, ordinal))
+		if err := r.CreateShoot(ctx, a.gardenClient.Client(), ordinal); err != nil {
+			return status, err
+		}
+		updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootReconcilingReason)
+
+		// Increment Replicas and NextReplicaNumber in status
+		status.Replicas++
+		status.NextReplicaNumber++
+
+		return status, nil
+
+	case scalingIn:
+		// Determine the replica to be deleted
+		// From all deletable replicas, choose the one with lowest priority
+		if len(deletableReplicas) == 0 {
+			return status, fmt.Errorf("no deletable replicas found")
+		}
+		sort.Sort(ascendingPriority(deletableReplicas))
+		r := deletableReplicas[0]
+
+		// Delete the replica's managed seed (if it exists), or its shoot (if not)
+		if replicaManagedSeedExists(r.GetStatus()) {
+			a.infoEventf(set, "Deleting managed seed %s", r.GetFullName())
+			if err := r.DeleteManagedSeed(ctx, a.gardenClient.Client()); err != nil {
+				return status, err
+			}
+			updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ManagedSeedDeletingReason)
+		} else {
+			a.infoEventf(set, "Deleting shoot %s", r.GetFullName())
+			if err := r.DeleteShoot(ctx, a.gardenClient.Client()); err != nil {
+				return status, err
+			}
+			updatePendingReplica(status, r.GetName(), seedmanagementv1alpha1.ShootDeletingReason)
+		}
+
+		// Decrement ReadyReplicas in status
+		if replicaIsReady(r) {
+			status.ReadyReplicas--
+		}
+
+		return status, nil
+	}
+
+	a.infoEventf(set, "All replicas ready")
+	status.PendingReplica = nil
+	return status, nil
+}
+
+func (a *actuator) infoEventf(set *seedmanagementv1alpha1.ManagedSeedSet, fmt string, args ...interface{}) {
+	a.recorder.Eventf(set, corev1.EventTypeNormal, gardencorev1beta1.EventReconciling, fmt, args...)
+	a.getLogger(set).Infof(fmt, args...)
+}
+
+func (a *actuator) getLogger(set *seedmanagementv1alpha1.ManagedSeedSet) *logrus.Entry {
+	return logger.NewFieldLogger(a.logger, "managedSeedSet", kutil.ObjectName(set))
+}
+
+func updatePendingReplica(status *seedmanagementv1alpha1.ManagedSeedSetStatus, name string, reason seedmanagementv1alpha1.PendingReplicaReason) {
+	if status.PendingReplica == nil || status.PendingReplica.Name != name || status.PendingReplica.Reason != reason {
+		status.PendingReplica = &seedmanagementv1alpha1.PendingReplica{
+			Name:   name,
+			Reason: reason,
+			Since:  Now(),
+		}
+	}
+}
+
+func replicaIsReady(r Replica) bool {
+	return r.GetStatus() == StatusManagedSeedRegistered && r.IsSeedReady() && r.GetShootHealthStatus() == operationshoot.StatusHealthy
+}
+
+func replicaDebugString(r Replica) string {
+	return fmt.Sprintf("%s:%s,%t,%s", r.GetFullName(), r.GetStatus().String(), r.IsSeedReady(), r.GetShootHealthStatus())
+}
+
+func replicaManagedSeedExists(status ReplicaStatus) bool {
+	return status == StatusManagedSeedPreparing || status == StatusManagedSeedDeleting || status == StatusManagedSeedRegistered
```

Comment:
If `StatusUnknown` would be the lowest status (`==0`), you could just do `status >= StatusManagedSeedPreparing` here.

Reply:
OK